packages/roosterjs-content-model/lib/formatHandlers/block/displayFormatHandler.ts

```diff
@@ -7,14 +7,23 @@ import { FormatHandler } from '../FormatHandler';
 export const displayFormatHandler: FormatHandler<DisplayFormat> = {
     parse: (format, element) => {
         const display = element.style.display;
+        const flexDirection = element.style.flexDirection;
 
         if (display) {
             format.display = display;
         }
+
+        if (flexDirection) {
```

Comment:
Try merge this into direction handler. if it has flexDirection and display is flex, use it, otherwise fallback to textAlign.